teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,707 @@
+      value: task/buildpacks/buildpacks.yaml
+    - name: branch
+      value: main
+```
+
+Note that a user can specify either `"commit"` or `"branch"`, but not both. We need to support `"tags"` as well as a
+form of revision. Therefore, we propose removing `"commit"` and `"branch"` from Git Resolver, and replace them
+with `"revision"` which can be a commit, branch or tag. The `"revision"` will default to `"main"` if not provided.
+
+```yaml
+# using commit in revision
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: git
+    resource:
+    - name: url
+      value: https://github.com/tekton/catalog.git
+    - name: path
+      value: task/buildpacks/buildpacks.yaml
+    - name: revision
+      value: d9183e36f00c712b5de59932650c071674ea89b8
+---
+# using branch in revision
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: git
+    resource:
+      - name: url
+        value: https://github.com/tekton/catalog.git
+      - name: path
+        value: task/buildpacks/buildpacks.yaml
+      - name: revision
+        value: main
+---
+# using tag in revision
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: git
+    resource:
+      - name: url
+        value: https://github.com/tekton/catalog.git
+      - name: path
+        value: task/buildpacks/buildpacks.yaml
+      - name: revision
+        value: v0.5 
+```
+
+#### Hub Resolver
+
+[Hub Resolver][hub-resolver] has a `"version"` parameter that can be used to get a specific version of a resource,
+for example [this version][bp-hub] of buildpacks `Task`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: hub
+    resource:
+    - name: kind
+      value: task
+    - name: name
+      value: buildpacks
+    - name: version
+      value: 0.5
+```
+
+The `"version"` will default to the latest version if not provided.
+
+We propose adding a `"catalog"` parameter to the Hub Resolver. This will be useful in resolving conflicts when
+resources from separate Catalogs have the same names. For example, a `TaskRun` executing the "openshift-client"
+from the Openshift Catalog described [above](#openshift) would use the Hub Resolver as such:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: openshift-install
+spec:
+  taskRef:
+    resolver: hub
+    resource:
+    - name: kind
+      value: task
+    - name: name
+      value: openshift-client
+    - name: catalog
+      value: openshift
+    - name: version
+      value: 0.3
+```
+
+#### HTTP Endpoint
+
+In [TEP-0110][tep-0110], we added an HTTP endpoint that serves Tekton resources that users can apply directly to their
+clusters. These endpoints should continue working as is:
+
+```shell
+kubectl apply -f https://hub.tekton.dev/tekton/task/buildpacks/0.5/raw
```

Comment:
just to avoid any confusion
```suggestion
kubectl apply -f https://api.hub.tekton.dev/tekton/task/buildpacks/0.5/raw
```

Reply:
good catch, fixed the link -- https://api.hub.tekton.dev/v1/resource/tekton/task/buildpacks/0.5/raw